docs/calendar.md

```diff
@@ -36,6 +36,46 @@ const calendar = new Intl.DateTimeFormat().resolvedOptions().calendar;
 date.withCalendar(calendar).add({ months: 1 });
 ```
 
+### Invariants Across Calendars
+
+The following "invariants" (statements that are always true) hold for all built-in calendars, and SHOULD hold for properly-authored custom calendars:
+
+- `year`, `month`, `day` are always integers that increase as time goes forward
+- `month` and `day` are always positive integers
+- `date.month === 1` during the first month of any year
+- `date.month === date.monthsInYear` during the last month of any year
+- `month` is always continuous (no gaps)
+- Any date can be serialized to an object using only four properties: `{ year, month, day, calendar }`
+
+### Writing Cross-Calendar Code
+
+Here are best practices for writing code that will work regardless of the calendar used:
+
+- Validate or coerce the calendar of all external input. If your code receives date data from an external source, you should validate that its calendar is what you expect or coerce it to a known calendar via the `withCalendar` method. Otherwise, a sophisticated attacker could change the behavior of your app or introduce security or performance issues by introducing an unexpected calendar.
```

Comment:
I think this might need some rephrasing. It's not likely that code would receive a Temporal object in an _unknown_ calendar, unless they had a dependency on a userland calendar somewhere in their stack. Probably what we want to advise people to guard against is _unexpected_ calendars.

How about this?
```suggestion
- Validate or coerce the calendar of all external input. If your code receives a Temporal object from an external source, you should check that its calendar is what you expect, and if you are not prepared to handle other calendars, convert it to the ISO calendar using `withCalendar('iso8601')`. Otherwise, you may end up with unexpected behavior in your app or introduce security or performance issues by introducing an unexpected calendar.
```